ci/infra/azure/README.md

```diff
@@ -0,0 +1,160 @@
+# Introduction
+
+This terraform project creates the infrastructure needed to run a
+cluster on top of Azure.
+
+Once the infrastructure is up and running nothing special has to be done
+to deploy CaaS Platform on top of it.
+
+This document focuses on the key aspects of the infrastructure created
+by terraform.
+
+# Cluster layout
+
+## Machines
+
+As usual the cluster is based on two types of nodes: master and worker nodes.
+
+All the nodes are created using the SLES 15 SP1 container host image built
+and maintained by the SUSE Public Cloud team.
+
+Right now users **must** bring their own license into the public cloud to be
+able to access SLE and SUSE CaaS Platform packages.
+
+The machines are automatically registered at boot time against SUSE Customer
+Center, RMT or a SUSE Manager instance depending on which one of the following
+variables has been set:
+
+  * `caasp_registry_code`
+  * `rmt_server_name`
+  * `suma_server_name`
+
+The SLES images [do not yet support cloud-init](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/using-cloud-init)
+(it will probably be supported starting from SLE15 SP2). In the meantime
+terraform leverages the Azure Linux Extension capabilities provided by
+the [Azure Linux Agent](https://docs.microsoft.com/en-us/azure/virtual-machines/extensions/agent-linux).
+
+### Using spot instances
+
+It's possible to create a [spot VMs](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/spot-vms)
+both for the master and worker nodes.
+
+This can be done by setting these variables to `true` (they are set to
+`false` by default):
+
+  * `master_use_spot_instance`
+  * `worker_use_spot_instance`
+
+## Network layout
```

Comment:
I would suggest to add some description about vnet peering support.

Reply:
added